tests/test_hexary_trie.py

```diff
@@ -109,3 +121,122 @@ def test_trie_using_fixtures(fixture_name, fixture):
         actual_root = trie.root_hash
 
         assert actual_root == expected_root
+
+
+class KeyAccessLogger(dict):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.read_keys = set()
+
+    def __getitem__(self, key):
+        result = super().__getitem__(key)
+        self.read_keys.add(key)
+        return result
+
+    def unread_keys(self):
+        return set(self) - self.read_keys
```

Comment:
Am I the only one that gets curmudgeonly when it comes to pythons default to iterate over just keys in dictionaries?

Reply:
Haha, I suppose at first it bothered me. I expected `.items()` to be the default. Now it's just like little quirks in the English language: I know they're there, but I don't think about them anymore.

I'm happy to be explicit here and use `self.keys()`, though.